### PR TITLE
Add tests for some commands

### LINKED ISSUE #99: Tests: Setup command

### LINKED ISSUE #127: Unable to execute setup without specifying Focus Area
Focus Area should be optional

### LINKED ISSUE #133: Tests: Delete command

### LINKED ISSUE #134: Tests: Goto command

### LINKED ISSUE #135: Tests: Find command

### PR DESCRIPTION
**Pull request by GabrielYik**
The commands are `delete`, `goto`, `setup` and `find`.
Resolves #99  
Resolves #133  
Resolves #134  
Resolves #135   

Make the focus areas parameter optional in the setup command. Resolves #127

### CODE REVIEW

File: src/test/java/seedu/planner/commons/util/StringUtilTest.java [hunk 1]
```diff
@@ -205,4 +207,47 @@ public void convertCollectionToString_nullGiven_throwsNullPointerException() {
...
+        assertTrue(areEqualIgnoreCase("Pathfinder", "Pathfinder"));
+
+        // Both difference case
+        assertTrue(areEqualIgnoreCase("Pathfinder", "PaThFiNdEr"));
+    }
+
+    @Test
+    public void areEqualIgnoreCase_nullGiven_throwsNullPointerException() {
+        thrown.expect(NullPointerException.class);
+        areEqualIgnoreCase(null, null);
```
Comment (Hilda-Ang) [inline src/test/java/seedu/planner/commons/util/StringUtilTest.java]: Can add if only one of them is `null` also

File: src/test/java/seedu/planner/logic/commands/DeleteCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    @Test
+    public void execute_invalidModule_throwsCommandException() {
+        Module moduleToDelete = INVALID_MODULE_CS0000;
+        DeleteCommand deleteCommand = new DeleteCommand(List.of(moduleToDelete));
+
+        String expectedMessage = String.format(Messages.MESSAGE_INVALID_MODULES, moduleToDelete);
+
+        assertCommandFailure(deleteCommand, model, commandHistory, expectedMessage);
+    }
+
```
Comment (Hilda-Ang) [inline src/test/java/seedu/planner/logic/commands/DeleteCommandTest.java]: Maybe can add test for module that is valid, but has been deleted or has not been added?

File: src/test/java/seedu/planner/logic/commands/GoToCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        assertEquals(expectedMessage, result.feedbackToUser);
+        assertTrue(eventsCollectorRule.eventsCollector.getMostRecent() instanceof TabSwitchEvent);
+        assertEquals(eventsCollectorRule.eventsCollector.getSize(), 1);
+    }
+
+    @Test
+    public void execute_invalidArgs_throwsCommandException() throws CommandException {
+        GoToCommand goToCommand = new GoToCommand(INVALID_YEAR_FIVE, INVALID_SEMESTER_THREE);
+
```
Comment (Hilda-Ang) [inline src/test/java/seedu/planner/logic/commands/GoToCommandTest.java]: Can add test for `GoToCommand` with valid year but invalid semester, vice versa

File: src/test/java/seedu/planner/logic/commands/SetUpCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                VALID_MAJOR_CS, Set.of(VALID_FOCUS_AREA_SE));
+
+        assertCommandSuccess(setUpCommand, model, commandHistory, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidArgs_throwsCommandException() {
+        SetUpCommand setUpCommand = new SetUpCommand(INVALID_YEAR_FIVE, VALID_SEMESTER_ONE,
+                VALID_MAJOR_CS, Set.of(VALID_FOCUS_AREA_SE));
+
```
Comment (Hilda-Ang) [inline src/test/java/seedu/planner/logic/commands/SetUpCommandTest.java]: Maybe add tests for other invalid parameters?

File: src/test/java/seedu/planner/logic/commands/SetUpCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        String expectedMessage = String.format(MESSAGE_SET_UP_SUCCESS,
+                VALID_YEAR_ONE, VALID_SEMESTER_ONE, VALID_MAJOR_CS, VALID_FOCUS_AREA_SE);
+        Model expectedModel = new ModelManager();
+        expectedModel.setUpUserProfile(VALID_YEAR_ONE, VALID_SEMESTER_ONE,
+                VALID_MAJOR_CS, Set.of(VALID_FOCUS_AREA_SE));
+
+        assertCommandSuccess(setUpCommand, model, commandHistory, expectedMessage, expectedModel);
+    }
+
```
Comment (Hilda-Ang) [inline src/test/java/seedu/planner/logic/commands/SetUpCommandTest.java]: Can add test for `SetUpCommand` with empty focus area too

File: src/test/java/seedu/planner/logic/commands/GoToCommandTest.java (hunk 3 above)
Comment (GabrielYik) [reply src/test/java/seedu/planner/logic/commands/GoToCommandTest.java]: Would valid year and invalid semester, and invalid year and valid semester, fall under the same EP?

Comment (GabrielYik) [reply src/test/java/seedu/planner/logic/commands/GoToCommandTest.java]: I think in this case, white box testing is being conducted. Internally, an `or` expression is used to evaluate the validity of the arguments. So one more test involving one invalid arg should be enough

Comment (Hilda-Ang) [reply src/test/java/seedu/planner/logic/commands/GoToCommandTest.java]: Okay then, one more should be enough
